server/model/puzzle.ts

```diff
@@ -37,11 +38,18 @@ export async function listPuzzles(
       SELECT pid, uploaded_at, content
       FROM puzzles
       WHERE is_public = true
+      AND (content->'info'->>'type') = ANY($1)
+      AND ((content -> 'info' ->> 'title') || (content->'info'->>'author')) ILIKE ALL($2)
       ORDER BY pid DESC 
```

Comment:
unrelated to this PR, but noticed while testing: pid is a string, so this sorts lexicographically. I'll change this sort by number (or by uploaded at) separately.